pallets/gear/src/lib.rs

```diff
@@ -326,91 +328,120 @@ pub mod pallet {
-                    }
-                    JournalNote::SendDispatch { dispatch, .. } => {
-                        gas_to_send =
-                            gas_to_send.saturating_add(dispatch.message.gas_limit().unwrap_or(0));
-                    }
-                    JournalNote::MessageDispatched(CoreDispatchOutcome::MessageTrap { .. }) => {
-                        return None;
+            let block_info = BlockInfo {
+                height: <frame_system::Pallet<T>>::block_number().unique_saturated_into(),
+                timestamp: <pallet_timestamp::Pallet<T>>::get().unique_saturated_into(),
+            };
+
+            let existential_deposit = T::Currency::minimum_balance().unique_saturated_into();
+
+            let mut max_gas_spent = 0;
+
+            while let Some(dispatch) = common::dequeue_dispatch() {
+                let actor_id = dispatch.message.dest;
+                let actor = ext_manager
+                    .get_executable_actor(actor_id)
+                    .ok_or_else(|| b"Program not found in the storage".to_vec())?;
+
+                let journal = core_processor::process::<
+                    ext::LazyPagesExt,
+                    SandboxEnvironment<ext::LazyPagesExt>,
+                >(
+                    Some(actor),
+                    dispatch.into_dispatch(initial_gas),
+                    block_info,
+                    existential_deposit,
+                    ProgramId::from_origin(source),
+                );
+
+                core_processor::handle_journal(journal.clone(), &mut ext_manager);
+
+                let (remaining_gas, _) =
+                    T::GasHandler::get_limit(root_message_id).ok_or_else(|| {
+                        b"Internal error: unable to get gas limit after execution".to_vec()
+                    })?;
+
+                // TODO: Check whether we charge gas fee for submitting code after #646
+                for note in journal {
+                    match note {
+                        JournalNote::SendDispatch { .. }
+                        | JournalNote::WaitDispatch(..)
+                        | JournalNote::MessageConsumed(..) => {
+                            let gas_spent = initial_gas.saturating_sub(remaining_gas);
+                            if gas_spent > max_gas_spent {
+                                max_gas_spent = gas_spent;
+                            }
+                        }
+                        JournalNote::MessageDispatched(CoreDispatchOutcome::MessageTrap {
+                            trap,
+                            ..
+                        }) => {
+                            return Err(format!(
+                                "Program terminated with a trap: {}",
+                                trap.unwrap_or("No reason")
+                            )
+                            .into_bytes());
+                        }
+                        _ => (),
```

Comment:
```suggestion
                        _ => {},
```

Reply:
Can't agree with it, the unit type at the empty branch is more idiomatic.

https://doc.rust-lang.org/stable/rust-by-example/flow_control/match/binding.html
https://doc.rust-lang.org/book/ch06-02-match.html#catch-all-patterns-and-the-_-placeholder